timesheets/src/Code.js

```diff
@@ -0,0 +1,129 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License.  You may obtain a copy
+// of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+// Global variables representing the index of certain columns.
+var COLUMN_NUMBER = {
+  EMAIL: 3,
+  HOURS_START: 4,
+  HOURS_END: 8,
+  HOURLY_PAY: 9,
+  CALC_PAY: 10,
+  APPROVAL: 11,
+};
+
+/** Creates the menu item "Timesheets" so user can run scripts on drop-down.
+*/
+function onOpen() {
+  var ui = SpreadsheetApp.getUi();
+  ui.createMenu('Timesheets')
+      .addItem('Column Setup', 'columnSetup')
+      .addItem('Send Emails', 'sendEmails')
+      .addToUi();
+}
+
+/** Runs the "Column Setup" script. Adds "WEEKLY PAY" column with calculated
+ * values using array formulas. Adds an "APPROVAL" column at the end of
+ * the sheet, containing drop-down menus to either approve/disapprove employee
+ * timesheets.  Adds a "NOTIFIED STATUS" column indicating whether or not an
+ * employee has yet been e mailed.
+*/
+function columnSetup() {
+  // defines variables
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var lastCol = sheet.getLastColumn();
+  var lastRow = sheet.getLastRow();
+  var frozenRows = sheet.getFrozenRows();
+  var beginningRow = frozenRows + 1;
+
+  // adds new calculate pay column
+  sheet.insertColumnAfter(COLUMN_NUMBER.HOURLY_PAY);
+  sheet.getRange(1, COLUMN_NUMBER.CALC_PAY).setValue('WEEKLY PAY');
+
+  // calculates weekly pay using array formulas
+  sheet.getRange(beginningRow, COLUMN_NUMBER.CALC_PAY, lastRow - frozenRows, 1)
```

Comment:
I think you only want to set the formula on the first row, not all the rows.

Reply:
When I set it to just the first row, it only outputs weekly pay value for the first row. Will read more about array formulas!

Reply:
I think you need to change the `SUM` range from `D2:H2` to `D2:H`.